<?xml version="1.0"?>
<launch>
  <arg name="device" value="0" />
  <arg name="width" value="1280" />
  <arg name="height" value="1024" />

  <node name="uvc_camera" pkg="libuvc_camera" type="camera_node" 
      output="screen">
    <!-- Parameters used to find the camera -->
    <param name="vendor" value="0x0"/>
    <param name="product" value="0x0"/>
    <param name="serial" value=""/>
    <!-- If the above parameters aren't unique, choose the first match: -->
    <param name="index" value="$(arg device)"/>

    <param name="scanning_mode" value="1" />
    <!-- Image size and type -->
    <param name="width" value="$(arg width)"/>
    <param name="height" value="$(arg height)"/>
    <!-- choose whichever uncompressed format the camera supports: -->
    <param name="video_mode" value="uncompressed"/>  <!-- or yuyv/nv12/jpeg -->
    <param name="frame_rate" value="5"/>

    <param name="timestamp_method" value="start"/>  <!-- start of frame -->
    <!--param name="camera_info_url" value="file:///tmp/cam.yaml"/-->

    <param name="auto_exposure" value="0"/>  <!-- use aperture_priority auto exposure -->
    <param name="auto_white_balance" value="false"/>
  </node>
  
  <!-- rqt_image_view doesn't support arg yet
  <node name="raw_image_view" pkg="rqt_image_view" type="rqt_image_view"
      args="image_raw" />
  -->
  <node name="raw_image_view" pkg="image_view" type="image_view"
      >
    <remap from="image" to="image_raw" />
  </node>

  <node name="image_deque" pkg="vimjay" type="image_deque" >
    <remap from="image" to="image_raw" />
  </node>

  <node name="captured_image_view" pkg="image_view" type="image_view"
      >
    <remap from="image" to="captured_image" />
  </node>
</launch>
